### ISSUE TITLE
Nextcloud regularly reports "Local file not found" on various backup files

### ISSUE THREAD

**Issue by eighthave** [seedvault-app/seedvault#266]
On CalyxOS, I went through SetupWizard and configured SeedVault to use my own Nextcloud.  I don't remember details from that process, but I can report settings from my device if you need them.  Now it occasionally reports "Local file not found" on various backup files "Lokale Datei nicht gefunden". My guess is that SeedVault is creating files, then Nextcloud sees them, then SeedVault deletes or renames them before Nextcloud runs the sync.  I have Nextcloud set to sync only when there is power+wifi, so basically once a day.  The only other time I get that message is when I take pictures on the phone, then delete them locally before Nextcloud syncs.

![signal-2021-06-24-075121](https://user-images.githubusercontent.com/143465/123226239-ff7c1d00-d4c2-11eb-9b8b-32b9f90b7c2b.png)

**Comment by chirayudesai** [comment]
What does it say for these apps under Settings -> System -> Backup -> Backup status

F-Droid, Jitsi Meet and others.

**Comment by eighthave** [comment]
Unfortunately, due to #267 my backup status was reset.  I'll report back next time this happens to me.

**Comment by grote** [comment]
>  I have Nextcloud set to sync only when there is power+wifi

Did you ever observe backups running? Do you know if this setting prevents Nextcloud from uploading them right away? Normally, you see a Seedvault notification about running backups and roughly at the same time a Nextcloud notification related to uploads.

**Comment by eighthave** [comment]
I haven't noticed whether the backup uploads right away.  My guess is that it does not upload immediately, but waits until WiFi+Power are plugged in.  That's the case for photos/videos at least.  I think only when I manually upload a file does it upload it regardless of the settings.  I know that these past three runs were while the device was not plugged into power:

![Screenshot_20210705-171743](https://user-images.githubusercontent.com/143465/124493285-f7ff2280-dda4-11eb-9080-131d1591db6f.png)


As for  Settings -> System -> Backup -> Backup status, it said ""about 9 hours ago":

![Screenshot_20210705-171850](https://user-images.githubusercontent.com/143465/124493217-e61d7f80-dda4-11eb-9dea-006198b10bf2.png)

**Comment by eighthave** [comment]
I triggered a backup with "Jetzt sichern" (backup now) in the settings while on WiFi but not plugged in.  It looks like it was working, I have a Seedvault notification saying "Sicherung läuft" (backup is running) and then a Nextcloud notification saying "Lade hoch" (uploading) and it looks like it is uploading files/dirs named after Application IDs of installed apps.

Then, when the Seedvault progress said it was something like 34% complete, I think the screen locked.  Then unlocking it, I got this ("Upload Error"):

![Screenshot_20210706-091947](https://user-images.githubusercontent.com/143465/124567554-a487e580-de33-11eb-8040-43ce35db2bc2.png)

It sat like that for a while, and Seedvault showed 34% on the progress on the notiifcation.  Then I let it sit for a while, and when I came back, the Seedvault notification said there was an error.

**Comment by grote** [comment]
I have occasional errors like that with Nextcloud as well. Though "locked" errors are rare. I already spend significant time improving the worst part of their SAF implementation. Not much more we can do about that. I still hope one day to have a dedicated WebDav/SSH/SFTP plugin to not having to go through the Nextcloud app.

**Comment by initstring** [comment]
I had the exact same issue, also on CalyxOS with Nextcloud connected and syncing normally. Lots of failed uploads every single time I manually triggered a backup or when a backup ran on schedule. I saw a bunch of the .apk files fail, retry, sometimes succeed. And I saw a lot of the "file no found" errors as well.

This was fixed by disabling battery optimisation for Nextcloud. Settings -> Apps  & Notifications -> Advanced -> Special app access -> Battery optimisation -> Nextcloud. Set to "not optimised".

@eighthave - perhaps your problem might be fixed the same way? You mention you "let it sit for a while" which is what I was doing as well. Possibly some power saving function was interfering.

Anyway, thank you so much to the Seedvault team for this amazing project!
